vendor/github.com/ishidawataru/sctp/sctp_unsupported.go

```diff
@@ -0,0 +1,46 @@
+// +build !linux linux,386
+
+package sctp
+
+import (
+	"errors"
+	"net"
+)
+
+var ErrUnsupported = errors.New("operation unsupported")
```

Comment:
This seems like a perfectly reasonable discipline for building cleanly on other platforms in general.  But from the docker/libnetwork perspective, I'm concerned about two things:
  * This error message is way too generic -- if libnetwork just returns this and the caller prints the string there's no context to say what isn't supported.  At the very least perhaps the string should be changed to "SCTP not supported on this platform" or some such.
   * This file is fairly deep in the call stack to be flagging lack of support as far as libnetwork goes.  Ideally either libnetwork or moby itself should flag early that an SCTP operation isn't supported by "this version" on "this platform".  It kind of seems wrong to, for example, try to create an SCTP proxy thinking this is perfectly valid up until we attempt to bind a listening socket.  Maybe I'm just being overly cautious on this one though.